### DISCUSSION TITLE
How to use workflow designer in elsa 2

### DISCUSSION THREAD

**Discussion by yariamirhossein** [elsa-workflows/elsa-core#488] [Q&A]
I'm using elsa 2.0.0-preview6.1080, and I want to use workflow designer but I can't run it. Please help...
My startup.cs is like below  :

`public void ConfigureServices(IServiceCollection services)
		{
			services.AddRazorPages();
			services
								.AddElsa(option => option.UsePersistence(conf => conf.UseSqlServer("Server=localhost;Database=WFTest01;User=sa;Password=Aa@123456;", IsolationLevel.ReadUncommitted)))
								.AddHttpActivities()
								.AddConsoleActivities()
								.AddDataMigration<Migrations>()
								.AddIndexProvider<DocumentIndexProvider>()
								.AddWorkflowContextProvider<DocumentWorkflowContextProvider>()
								.AddWorkflow<DocumentApprovalWorkflow>()
								.AddElsaApiEndpoints()
			.AddElsaClient(conifgure =>
			{
				conifgure.ServerUrl = new Uri("https://localhost:44319/");
			});
			
		}

		// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
		public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
		{
			if (env.IsDevelopment())
			{
				app.UseDeveloperExceptionPage();
			}
			else
			{
				app.UseExceptionHandler("/Error");

				app.UseHsts();
			}
			// Add HTTP activities middleware.
			app.UseHttpActivities();

			// Show welcome page if no routes matched any endpoints.
			//app.UseWelcomePage();

			app.UseHttpsRedirection();
			app.UseStaticFiles();
			app.UseRouting();

			app.UseEndpoints(endpoints =>
			{
				endpoints.MapRazorPages();
			});

			//app.UseEndpoints(endpoints => endpoints.MapControllers());

		}`
I don't know what extra configuration I should do or which url I should try to see designer.
thank you 👍

**Comment by sfmskywalker** [comment]
Elsa 2 has no designer at the moment (although it is [currently under development](https://github.com/elsa-workflows/elsa-core/raw/feature/elsa-2.0/doc/elsa-2-dashboard-plus-designer.gif)).

The only way to use Elsa 2 preview as of this writing is by implementing them via code (see the [samples](https://github.com/elsa-workflows/elsa-core/tree/feature/elsa-2.0/src/samples) directory)

**Comment by yariamirhossein** [comment]
Thank you so much for your response. 👍 
Your new design is great 🥇

**Comment by sfmskywalker** [comment]
You're welcome. Thank you :)

**Comment by sfmskywalker** [comment]
The original 2020 answer was correct for `2.0.0-preview6`: there was no designer available in that preview yet.

For current Elsa, the visual designer lives in the separate `elsa-studio` repository rather than inside `elsa-core`. In the current Studio code, `AddWorkflowsModule()` wires in the workflows feature, default activity port providers, and `AddWorkflowsDesigner()` (`elsa-studio/src/modules/Elsa.Studio.Workflows/Extensions/ServiceCollectionExtensions.cs:33-56`), and the sample WASM host registers that module directly (`elsa-studio/src/hosts/Elsa.Studio.Host.Wasm/Program.cs:92-105`).

So for modern versions, the path is to run/use Elsa Studio (or one of the Studio hosts / embeddable components), not to look for a designer URL inside the old preview server setup shown in the original post.

Since the preview-era question was already answered and acknowledged by the original poster, I'm closing this discussion as resolved.
